### ISSUE TITLE
A strange example

### ISSUE THREAD

**Issue by zzhisthebest** [marcusrossel/lean-egg#43]
```
example {a b c : ℕ} : 2*a=a+a :=by
--proof 1,success
  -- rw [←two_mul]
--proof 2,egg failed to prove the goal (saturated)
  egg [two_mul]
```
Why this happened?

**Comment by marcusrossel** [comment]
This problem is related to the fact that `two_mul.{u} {α : Type u} [NonAssocSemiring α] (n : α) : 2 * n = n + n` is a lemma about any type `α` which has a `NonAssocSemiring α` instance, instead of just `ℕ`. We're aware that this doesn't work yet and are planning to investigate this in the near future.

**Comment by goens** [comment]
@marcusrossel with the new `egg.eraseTCinstances` this seems to sort of work, but reconstruction fails, we get:
```
egg failed to build proof: initial expression is not defeq to lhs of proof goal
```
Looking at it, the typeclass synthesis builds for the LHS the following `HMul` instance:
```
@instHMul ℕ NonUnitalNonAssocSemiring.toMul
```

But that doesn't exist for `Nat`:
```
#synth NonUnitalNonAssocSemiring ℕ
-- failed to synthesize
--  NonUnitalNonAssocSemiring ℕ
```

I wonder why it's synthesizing that instance of `HMul`

**Comment by marcusrossel** [comment]
Yeah, `egg.eraseTCinstances` isn't complete, yet. In particular, I haven't touched the proof reconstruction part. So I'll reconsider this issue when proper proof reconstruction is in place.

**Comment by marcusrossel** [comment]
@zzhisthebest your example works now :)
